### ISSUE TITLE
ipset state check problem

### ISSUE THREAD

**Issue by skjaro** [saltstack/salt#44298]
### Description of Issue/Question
Hi,

I have problem with ipset states. It throws error when I try add multiple entries with type ip,mac type.
### Setup
```
mac_ip_allowed:
  ipset.set_present:
    - set_type: bitmap:ip,mac
    - range: 192.168.111.0/24

mac_ip_addr_allowed_entries:
  ipset.present:
    - set_name: mac_ip_allowed
    - entry:
       - 192.168.111.22,01:02:03:04:05:06
       - 192.168.111.23,01:02:03:04:05:07
```
### Steps to Reproduce Issue
```
          ID: mac_ip_addr_allowed_entries
    Function: ipset.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1843, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1795, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/ipset.py", line 211, in present
                  family) is True:
                File "/usr/lib/python2.7/site-packages/salt/modules/ipset.py", line 431, in check
                  current_members = _parse_members(settype, _find_set_members(set))
                File "/usr/lib/python2.7/site-packages/salt/modules/ipset.py", line 582, in _parse_members
                  return [_parse_member(settype, member) for member in members]
                File "/usr/lib/python2.7/site-packages/salt/modules/ipset.py", line 593, in _parse_member
                  part = parts[i]
              IndexError: list index out of range
     Started: 14:40:57.583261
    Duration: 24.065 ms
```

### Versions Report
```
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.2.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

I think the problem is in this lines in "/usr/lib/python2.7/site-packages/salt/modules/ipset.py":
```
    subtypes = settype.split(':')[1].split(',')

    parts = member.split(' ')

    parsed_member = []
```
should be:
```
    subtypes = settype.split(':')[1].split(',')

    parts = member.split(',')

    parsed_member = []
```

There is smiliar problem with module in #39552 issue.

**Comment by gtmanfred** [comment]
@skjaro Would you mind submitting this change as a PR? We would greatly appreciate it.

@lingonl can you take at this?

Thanks,
Daniel

**Comment by lingonl** [comment]
Yeah, I'll look into it.

**Comment by skjaro** [comment]
@gtmanfred Yes, if I have little time I submit it as PR. I look in code of module and it needs little rework, I have in mind some optimizations of module for many entries, should it read from file and not execute ipset command for each entry.

**Comment by gtmanfred** [comment]
Thanks!

**Comment by skjaro** [comment]
I have submitted change as PR. It was taken from #39552 issue becouse my fix did not take care on options with ipset entries. I thik @Xiami2012 done it right.
Module ipset need some rework becouse newer ipset has another options like:
[ skbmark value ] [ skbprio value ] [ skbqueue value ] 
I think adding some posibility to restore rules could be fine.

**Comment by rallytime** [comment]
Closed via #44356

**Comment by skjaro** [comment]
@gtmanfred I have patch for skbprio options. I am testing it and submit it soon. I think very interesting functionality in ipset. It is in ipset from 2014 year, but I look at it now. Can be very helpful in situation with many entries and traffic shaping and I think it obsoletes hash tables in tc command, but ipset module in salt needs some optimizations in check function. I rewrite it soon. There are another types in ipset such as hash:mac.
